### ISSUE TITLE
Add examples to the doc folder

### ISSUE THREAD

**Issue by lara** [artichoke/artichoke#367]
I have a simple Ruby app that I would like to compile to WebAssembly or just run on Artichoke for fun/benchmarking.

It would be really helpful if you could add some instructions on how to do that after setting up a build environment.

**Comment by lopopolo** [comment]
You're right, we do need docs. Sorry this isn't easier today. Below is my best attempt at the steps to get started. Let me know if they work for you.

The best example of how to build an app is the [spec-runner crate](https://github.com/artichoke/artichoke/blob/763c7d50b3f4e59f3d19f47fc0554a8014e4c7ea/spec-runner).

The steps are:

1. Create a new crate.

        cargo new [package name]

2. Add the `rust-embed` crate to the crate dependencies. `rust-embed` enables us to easily embed Ruby sources into the project.

https://github.com/artichoke/artichoke/blob/763c7d50b3f4e59f3d19f47fc0554a8014e4c7ea/spec-runner/Cargo.toml#L15-L17

3. Embed Ruby sources into crate.

https://github.com/artichoke/artichoke/blob/763c7d50b3f4e59f3d19f47fc0554a8014e4c7ea/spec-runner/src/mspec.rs#L16-L18

4. Load Ruby sources into an interpreter.

https://github.com/artichoke/artichoke/blob/763c7d50b3f4e59f3d19f47fc0554a8014e4c7ea/spec-runner/src/mspec.rs#L8-L14

5. Execute the Ruby entry point in `main`.

https://github.com/artichoke/artichoke/blob/763c7d50b3f4e59f3d19f47fc0554a8014e4c7ea/spec-runner/src/main.rs#L14-L54

For WebAssembly, you'll need the best example is the [Artichoke playground](https://github.com/artichoke/playground). You'll need to:

1. Depend on `stdweb` and wire up the event loop in `main`.
2. Expose functions to JS with FFI that initialize the interpreter and call the Ruby entrypoint.
3. Ensure these functions get exported with Emscripten.
4. Wire things up in JS with webpack.

**Comment by abinoam** [comment]
Hi @lopopolo,

I followed along your instructions, looking at mspec code in artichoke and updating according and I was able to embed a fragment of Ruby code into a binary.

The code is on https://github.com/abinoam/binrubyscript
And a blog post at https://abinoam.tl1n.com/embedding-ruby-code-in-an-executable-using-artichoke/

Congratulations for Artichoke!
